### PR TITLE
feat: New yaml file for updated CNI and CNS version

### PR DESCRIPTION
**Pull request by chengch801214**
**Reason for Change**:
Need an update to CNI version to resolve issue with newer kernels

**Issue Fixed**:

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by tamilmani1989 — DISMISSED** [review]

**Review by paulyufan2 — DISMISSED** [review]

File: .pipelines/mdnc/azure-cns-cni-1.5.28.yaml [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      tolerations:
+        - key: CriticalAddonsOnly
+          operator: Exists
+        - operator: "Exists"
+          effect: NoExecute
+        - operator: "Exists"
+          effect: NoSchedule
+      initContainers:
+        - name: init-cni-dropgz
+          image: "mcr.microsoft.com/containernetworking/cni-dropgz:v0.1.4" # CNI 1.5.28
```
Comment (rbtr) [inline .pipelines/mdnc/azure-cns-cni-1.5.28.yaml:99]:
```suggestion
          image: "mcr.microsoft.com/containernetworking/azure-cni:v1.5.28"
```

File: .pipelines/mdnc/azure-cns-cni-1.5.28.yaml [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+          command: ["/dropgz"]
+          args: ["deploy" , "azure-vnet", "-o", "/opt/cni/bin/azure-vnet", "azure-vnet-telemetry", "-o", "/opt/cni/bin/azure-vnet-telemetry", "azure-swift.conflist", "-o", "/etc/cni/net.d/10-azure.conflist"]
+          volumeMounts:
+            - name: cni-bin
+              mountPath: /opt/cni/bin
+            - name: cni-conflist
+              mountPath: /etc/cni/net.d
+      containers:
+        - name: cns-container
+          image: mcr.microsoft.com/containernetworking/azure-cns:v1.5.26
```
Comment (rbtr) [inline .pipelines/mdnc/azure-cns-cni-1.5.28.yaml:110]: why not bump both to 1.5.28?

**Review by rbtr — DISMISSED** [review]

File: .pipelines/mdnc/azure-cns-cni-1.5.28.yaml (hunk 2 above)
Comment (chengch801214) [reply .pipelines/mdnc/azure-cns-cni-1.5.28.yaml:110]: I believe the 1.5.28 CNS image is not available. Do the CNI and CNS versions have to match?

File: .pipelines/mdnc/azure-cns-cni-1.5.28.yaml (hunk 1 above)
Comment (chengch801214) [reply .pipelines/mdnc/azure-cns-cni-1.5.28.yaml:99]: Is the dropgz approach no longer recommended?

**Comment by chengch801214** [comment]
> Are the other manifests in this directory still in use? Is this strictly a version bump? Can we just write a kustomize template and let the consumer set these versions instead of hosting a manifest copy for every release?

Yes the others are still in use. Two of them are the ones we communicated to OpenAI to use, but our tests no longer work with them, so we created two other ones with updated CNI version for two different tests.

File: .pipelines/mdnc/azure-cns-cni-1.5.28.yaml (hunk 2 above)
Comment (rbtr) [reply .pipelines/mdnc/azure-cns-cni-1.5.28.yaml:110]: CNI and CNS are always released together, so if there is a 1.5.28 of CNI, there is for CNS also.
They don't have to match, but if you don't have a specific reason that you are holding one back for, might as well keep them together.

File: .pipelines/mdnc/azure-cns-cni-1.5.28.yaml (hunk 1 above)
Comment (rbtr) [reply .pipelines/mdnc/azure-cns-cni-1.5.28.yaml:99]: Correct. There is no functional difference though, this should be a drop-in replacement.

Comment (chengch801214) [reply .pipelines/mdnc/azure-cns-cni-1.5.28.yaml:99]: What about the /dropgz command? This init container has `command: ["/dropgz"]`, what would I replace that with if I use the azure-cni image directly?

Comment (chengch801214) [reply .pipelines/mdnc/azure-cns-cni-1.5.28.yaml]: Updated to use azure-cni:v1.5.28 directly.

File: .pipelines/mdnc/azure-cns-cni-1.5.28.yaml (hunk 2 above)
Comment (chengch801214) [reply .pipelines/mdnc/azure-cns-cni-1.5.28.yaml]: Updated to use CNS 1.5.28

**Review by rohitams — APPROVED** [review]

**Comment by thomasricci** [comment]
> Are the other manifests in this directory still in use? Is this strictly a version bump? Can we just write a kustomize template and let the consumer set these versions instead of hosting a manifest copy for every release?

yeah I think this makes sense. Otherwise we keep creating new files just to bump up versions

**Comment by chengch801214** [comment]
> > Are the other manifests in this directory still in use? Is this strictly a version bump? Can we just write a kustomize template and let the consumer set these versions instead of hosting a manifest copy for every release?
> 
> yeah I think this makes sense. Otherwise we keep creating new files just to bump up versions. These files are purely consumer by our e2e pipeline and nobody else should be using them.

I'm going to merge this PR first to unblock our E2E test. Having a customizable way to input the version is useful and can be done next time we need an update.

**Pull request merged**